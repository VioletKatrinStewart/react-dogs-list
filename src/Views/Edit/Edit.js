import React, { useState, useEffect } from 'react';
import Header from '../../Components/Header/Header';
import Form from '../../Components/Form/Form';
import { getDog, updateDog } from '../../services/DogData';
import { useParams } from 'react-router';
import { deleteDog } from '../../services/DogData';

export default function Edit() {
  const [name, setName] = useState('');
  const [bio, setBio] = useState('');
  const [age, setAge] = useState('');
  const [image, setImage] = useState('');
  const [breed, setBreed] = useState('');
  const [message, setMessage] = useState('');
  const [loading, setLoading] = useState(true);

  const params = useParams();
  useEffect(() => {
    const fetchData = async () => {
      const resp = await getDog(params.id);

      setName(resp.name);
      setBio(resp.bio);
      setImage(resp.image);
      setAge(resp.age);
      setBreed(resp.breed);
      setLoading(false);
    };
    fetchData();
  }, [params.id]);

  function handleChange(key, value) {
    if (key === 'name') {
      setName(value);
    } else if (key === 'bio') {
      setBio(value);
    } else if (key === 'age') {
      setAge(value);
    } else if (key === 'image') {
      setImage(value);
    } else if (key === 'breed') {
      setBreed(value);
    }
  }

  async function handleSave(e) {
    try {
      e.preventDefault();
      const dog = { name: name, age: age, bio: bio, image: image, breed: breed };
      await updateDog(dog, params.id);
      setMessage('Edited');
    } catch {
      setMessage('Error');
    }
  }

  async function handleDelete(e) {
    e.preventDefault();
    await deleteDog(params.id);
  }
  if (loading) return <h1>loading</h1>;

  return (
    <div>
      <Header></Header>
      <Form
        message={message}
        handleSave={handleSave}
        handleChange={handleChange}
        handleDelete={handleDelete}
        name={name}
        bio={bio}
        age={age}
        image={image}
        breed={breed}
      ></Form>
    </div>
  );
}
