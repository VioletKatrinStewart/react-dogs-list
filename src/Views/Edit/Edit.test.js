import { MemoryRouter, Route } from 'react-router-dom';
import Edit from './Edit';
import { render, screen } from '@testing-library/react';

it('should render the dogs info in the form', async () => {
  const { container } = render(
    <MemoryRouter initialEntries={['/dogs/6/edit']}>
      <Route path="/dogs/:id/edit">
        <Edit />
      </Route>
    </MemoryRouter>
  );
  //   screen.findByText('Name');
  await screen.findByDisplayValue('I am not a dog, how did I get here?');

  expect(container).toMatchSnapshot();
});
