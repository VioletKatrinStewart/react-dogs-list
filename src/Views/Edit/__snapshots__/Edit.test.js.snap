// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render the dogs info in the form 1`] = `
<div>
  <div>
    <header>
      <a
        href="/"
      >
        Home
      </a>
      <a
        href="/admin"
      >
        Admin
      </a>
    </header>
    <div
      class="form"
    >
      <h2>
         Enter the details to add a dog 
      </h2>
      <p />
      <label>
        Name
        <input
          placeholder="Name"
          type="text"
          value="Mittens "
        />
      </label>
      <label>
        Bio
        <input
          placeholder="Bio"
          type="text"
          value="I am not a dog, how did I get here?"
        />
      </label>
      <label>
        Age
        <input
          placeholder="Age"
          type="number"
          value=" 12"
        />
      </label>
      <label>
        Image
        <input
          placeholder="image"
          type="text"
          value="https://i.insider.com/5cdeb255021b4c06ca38bbb3?width=2000&format=jpeg&auto=webp"
        />
      </label>
      <label>
        Breed
        <input
          placeholder="Breed"
          type="text"
          value="Cat"
        />
      </label>
      <button>
        Submit
      </button>
    </div>
  </div>
</div>
`;
